eth/beacon/helpers.py

```diff
@@ -1,17 +1,167 @@
 from typing import (
     Iterable,
+    Any,
     List,
+    Tuple,
     TYPE_CHECKING,
 )
 
 from eth_utils import (
     to_tuple,
 )
 
+from eth_typing import (
+    Hash32,
+)
+
+from eth.utils.blake import (
+    blake,
+)
+from eth.beacon.types.shard_and_committee import (
+    ShardAndCommittee,
+)
+
+
 if TYPE_CHECKING:
+    from eth.beacon.config import BeaconConfig  # noqa: F401
+    from eth.beacon.types.active_state import ActiveState  # noqa: F401
+    from eth.beacon.types.attestation_record import AttestationRecord  # noqa: F401
+    from eth.beacon.types.block import Block  # noqa: F401
+    from eth.beacon.types.crystallized_state import CrystallizedState  # noqa: F401
     from eth.beacon.types.validator_record import ValidatorRecord  # noqa: F401
 
 
+#
+# Get block hashes
+#
+def get_block_hash(
+        active_state: 'ActiveState',
+        current_block: 'Block',
+        slot: int,
+        beacon_config: 'BeaconConfig') -> Hash32:
+    """
+    Get the blockhash from active_state.recent_block_hashes by current block slot number.
+    """
+    cycle_length = beacon_config.cycle_length
+
+    sback = current_block.slot_number - cycle_length * 2
+    assert sback <= slot < sback + cycle_length * 2
```

Comment:
This should be converted to an `if/raise` (we avoid assertions in runtime code because they are easy to disable)